### ISSUE TITLE
Comment attribution / printing -- odd example to investigate?

### ISSUE THREAD

**Issue by MysterAitch** [javaparser/javaparser#2539]
Here's an odd case of comment attribution where the comment seems to be attributed to multiple nodes thus ends up getting duplicated when printed. Specifically, note the duplicated FIXME comment.

Additionally, I would normally expect the `// list` comment to be attributed to the next element (i.e. the call to `conditional()` but it seems to be printed after the call to `child()`?

I've not yet dug into this to figure out what's actually going on (e.g. if it's an attribution versus printing issue) and would welcome some thoughts!

```java
        concreteSyntaxModelByClass.put(VariableDeclarator.class, sequence(comment(), child(ObservableProperty.NAME), // FIXME: we should introduce a derived property
        // list(ObservableProperty.EXTRA_ARRAY_LEVELS),
        conditional(ObservableProperty.INITIALIZER, IS_PRESENT, sequence(space(), token(GeneratedJavaParserConstants.ASSIGN), space(), child(ObservableProperty.INITIALIZER)))));
```

```java
        concreteSyntaxModelByClass.put(VariableDeclarator.class, sequence(// FIXME: we should introduce a derived property
        comment(), // FIXME: we should introduce a derived property
        child(ObservableProperty.NAME), // list(ObservableProperty.EXTRA_ARRAY_LEVELS),
        conditional(ObservableProperty.INITIALIZER, IS_PRESENT, sequence(space(), token(GeneratedJavaParserConstants.ASSIGN), space(), child(ObservableProperty.INITIALIZER)))));
```

**Comment by matozoid** [comment]
I can say that comment support in the lex pres printer is fiddly and is simply waiting for more bug reports in its current state.

**Comment by DiogoTheCoder** [comment]
Facing similar issues as well, https://github.com/javaparser/javaparser/issues/3042

**Comment by jlerbsc** [comment]
It seems to be related to CommentsInserter.attributeLineCommentsOnSameLine that sets the comment on all child nodes in the same line. This issue seems to be easy to solve but we have to define what is the expected result.

**Comment by jlerbsc** [comment]
This problem seems fixed in JavaParser 3.24.10.
